metaflow/plugins/events_decorator.py

```diff
@@ -0,0 +1,298 @@
+        elif self.attributes["flow"]:
+            # flow supports the format @trigger_on_finish(flow='FooFlow')
+            if is_stringish(self.attributes["flow"]):
+                self.triggers.append(
+                    {
+                        "fq_name": self.attributes["flow"],
+                    }
+                )
+            else:
+                raise MetaflowException(
+                    "Incorrect type for *flow* attribute in *@trigger_on_finish* "
+                    " decorator. Supported type is string - \n"
+                    "@trigger_on_finish(flow='FooFlow')"
+                )
+        elif self.attributes["flows"]:
+            # flows attribute supports the following formats -
+            #     1. flows=['FooFlow', 'BarFlow']
+            if isinstance(self.attributes["flows"], list):
+                for flow in self.attributes["flows"]:
+                    if is_stringish(flow):
+                        self.triggers.append(
+                            {
+                                "fq_name": flow,
+                            }
+                        )
+                    else:
+                        raise MetaflowException(
+                            "One or more flows in *flows* attribute in "
+                            "*@trigger_on_finish* decorator have an incorrect type. "
+                            "Supported type is string - \n"
+                            "@trigger_on_finish(flows=['FooFlow', 'BarFlow']"
+                        )
+            else:
+                raise MetaflowException(
+                    "Incorrect type for *flows* attribute in *@trigger_on_finish* "
+                    "decorator. Supported type is list - \n"
+                    "@trigger_on_finish(flows=['FooFlow', 'BarFlow']"
+                )
+
+        if not self.triggers:
+            raise MetaflowException(
+                "No flow(s) specified in *@trigger_on_finish* decorator."
+            )
+
+        # Make triggers @project aware
+        for trigger in self.triggers:
+            if trigger["fq_name"].count(".") == 0:
+                # fully qualified name is just the flow name
+                trigger["flow"] = trigger["fq_name"]
+            elif trigger["fq_name"].count(".") in (2, 3):
+                # fully qualified name is of the format - project.branch.flow_name
+                trigger["project"], _ = trigger["fq_name"].split(".", maxsplit=1)
+                trigger["branch"], trigger["flow"] = _.rsplit(".", maxsplit=1)
+            else:
+                raise MetaflowException(
+                    "Incorrect format for *flow* in *@trigger_on_finish* "
+                    "decorator. Specify either just the *flow_name* or a fully "
+                    "qualified name like *project_name.branch_name.flow_name*."
+                )
+            # TODO: Also sanity check project and branch names
+            if not re.match(r"^[A-Za-z0-9_]+$", trigger["flow"]):
```

Comment:
project and branch names have a different constraint (just noting here for future). We could/should import the re constants from there and check.

Reply:
Yes - already noted as a future TODO.